apps/ecommerce/lambda/src/routers/shopifyRouter.spec.ts

```diff
@@ -0,0 +1,118 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import { mockResourceLink } from '../mocks/resourceLink.mock';
+import Sinon from 'sinon';
+import Client, { CheckoutResource, CollectionResource, ImageResource, Product, ProductResource, Shop, ShopResource } from 'shopify-buy';
+
+const sandbox = Sinon.createSandbox();
+chai.use(chaiHttp);
+chai.should();
+
+const shopifyClientStub = {
+    product: {
+        fetch: () => Sinon.promise((resolve) => {
+            return resolve({
+                title: 'Some Product title',
+                description: 'Some product description',
+                images: [{ url: 'some-product-url' }]
+            } as Product)
+        }),
+        fetchAll: () => Sinon.promise(Sinon.spy()),
+        fetchMultiple: () => Sinon.promise(Sinon.spy()),
+        fetchByHandle: () => Sinon.promise(Sinon.spy()),
+        fetchQuery: () => Sinon.promise(Sinon.spy()),
+        fetchRecommendations: () => Sinon.promise(Sinon.spy()),
+        graphQLClient: null
+    } as ProductResource,
+    collection: {} as CollectionResource,
+    checkout: {} as CheckoutResource,
+    image: {} as ImageResource,
+    shop: {
+        fetchInfo: () => {
+            return Sinon.promise((resolve) => {
+                return resolve({
+                    id: 'some-shop-id',
+                    name: 'some test shop name'
+                } as Shop)
+            });
+        },
+        fetchPolicies: () => Sinon.promise(Sinon.spy()),
+        graphQLClient: null
+    } as ShopResource,
+    graphQLClient: null,
+    fetchNextPage: function <T extends Client.Node>(): Promise<T[]> {
+        throw new Error('Function not implemented.');
+    }
+}
+
+describe('Shopify Router', () => {
+    beforeEach((done) => {
+        sandbox.stub(NodeAppsToolkit, 'verifyRequest').get(() => {
+            return () => true;
+        });
+        done();
```

Comment:
I really don't want to poke the hornet's nest that is our async testing woes, but do you need `done` for these methods that aren't async?

Reply:
Good point. I did try to remove these `done()`'s, and it caused timeouts when testing. I also tried removing `done()`'s elsewhere in the file and ran into timeout errors. It seems like the nature of the sinon sandbox is a lot of async calls, so these are needed to avoid timeouts. Good suggestion though 👍